src/Bbt.Maybe.Tests/MaybeStructTests.cs

```diff
@@ -0,0 +1,154 @@
+﻿namespace Bbt.Maybe.Tests
+{
+    using Bbt.Maybe;
+    using Bbt.Maybe.Tests.TestData;
+    using Shouldly;
+    using Xunit;
+
+    public sealed class MaybeStructTests
+    {
+        public sealed class TheSomeMethod
+        {
+            [Fact]
+            public void ProjectToReferencingClass_ShouldReturnMaybeReferencingClass()
+            {
+                // Arrange
+                var lReferencedStruct = default(ReferencedStruct);
+                var lReferencingStruct = default(BaseStruct);
+                lReferencedStruct.ReferencingStruct = lReferencingStruct;
+
+                // Act
+                var lMaybeReferencingStruct = Maybe.SomeStruct<ReferencedStruct>(lReferencedStruct)
+                    .Some<BaseStruct>(aX => aX.ReferencingStruct);
+
+                // Assert
+                Maybe.SomeStruct<BaseStruct>(lReferencingStruct).ShouldBe(lMaybeReferencingStruct);
+            }
+
+            [Fact]
+            public void None_ShouldReturnMaybeNone()
+            {
+                // Arrange
+                var lMaybeReferencedStruct = Maybe.NoneStruct<ReferencedStruct>();
+
+                // Act
+                var lMaybeReferencingStruct = lMaybeReferencedStruct.Some<BaseStruct>(aX => aX.ReferencingStruct);
+
+                // Assert
+                Maybe.NoneStruct<BaseStruct>().ShouldBe(lMaybeReferencingStruct);
+            }
+        }
+
+        public sealed class TheDoMethod
+        {
+            [Fact]
+            public void MaybeValueNotSet_ActionNotCalled()
```

Comment:
In other projects we use a more natural style for naming test methods (e.g. `Action_Should_Not_Be_Called_If_Maybe_Value_Is_Not_Set()`). But can be cleaned up later

Reply:
Created #14 for tracking this